pkg/controller/replicaset/replica_set.go

```diff
@@ -193,6 +194,41 @@ func (rsc *ReplicaSetController) Run(workers int, stopCh <-chan struct{}) {
 	<-stopCh
 }
 
+// getReplicaSetsWithSameController returns a list of ReplicaSets with the same
+// owner as the given ReplicaSet.
+func (rsc *ReplicaSetController) getReplicaSetsWithSameController(rs *apps.ReplicaSet) []*apps.ReplicaSet {
+	controllerRef := metav1.GetControllerOf(rs)
+	if controllerRef == nil {
+		utilruntime.HandleError(fmt.Errorf("ReplicaSet has no controller: %v", rs))
+		return nil
+	}
+
+	allRSs, err := rsc.rsLister.ReplicaSets(rs.Namespace).List(labels.Everything())
+	if err != nil {
+		utilruntime.HandleError(err)
+		return nil
+	}
+
+	var relatedRSs []*apps.ReplicaSet
+	for _, r := range allRSs {
+		if ref := metav1.GetControllerOf(r); ref != nil && ref.UID == controllerRef.UID {
+			relatedRSs = append(relatedRSs, r)
+		}
+	}
+
+	var related string
+	if len(relatedRSs) > 0 {
+		var relatedNames []string
+		for _, r := range relatedRSs {
+			relatedNames = append(relatedNames, r.Name)
+		}
+		related = ": " + strings.Join(relatedNames, ", ")
```

Comment:
You should only be doing this inside of a klog.V(2) block like:

```
if klog.V(2) {
  var related string
  ...
  klog.Infof("Found ...", ...)
}

return relatedRSs
```

Reply:
Fixed.